public/docs/ts/latest/guide/component-communication.jade

```diff
@@ -0,0 +1,707 @@
+  same `MessageBus` instance. When instantiating `HeroPanel`, the framework will use the `MessageBus` instance injected into `HeroJobBoard`, 
+  because of the parent-child relationship between `HeroJobBoard` and `HeroPanel`.
+  
+.alert.is-important
+  :marked
+    We must not include `MessageBus` in the `providers` of `HeroJobBoard`, because it would break the logic. In that case `MessageBoard` would listen
+    to a separate `MessageBus` the other components are talking to. This is how the correct setup of `HeroJobBoard` is:
+
++makeExample('component-communication/ts/fragments/no-message-bus-in-hero-job-board.ts', null, 'message-board/hero-job-board.ts (extract)', {blk: /(providers: \[JobService\])/g})
+
+:marked
+  Now, we put together all pieces to check how the message bus works:
+  
+figure.image-display
+  img(src="/resources/images/devguide/component-communication/message-board-ui.png" alt="MessageBoard in action")
+  
+:marked
+  ## #6: Parent to child communication where children are transposed in &mdash; using `@ContentChildren`
+  
+  There are situations, when a parent component cannot determine the set of its child components on its own, because the collection of its children is managed
+  by an external entity. However, even in this case, the parent needs to communicate with each children. How can the parent establish the communication channel 
+  to its children? Angular provides several mechanism to achieve this, here we show the usage of the `@ContentChildren` annotation.
+  
+  In the previous scenarios, the `HeroJobBoard` component managed the collection of heroes to invite this way:
+  
++makeExample('component-communication/ts/fragments/heroes-to-invite.ts', null, null, {pnk: /(invitedHeroes)/g, blk: /(Hero\.heroes)/g})
+
+:marked
+  `HeroJobBoard` obtained the collection of heroes from the `heroes` static property of the `Hero` class. This solution was pretty useful for demonstration,
+  but it does not refelect the real life. In a real application, the list of heroes is probably retrieved from the backend.
+  
+  Let's play with the idea &mdash; and it's seems realistic &mdash; that the list of heroes to invite is managed outside of `HeroJobBoard`, and even it
+  changes while we use the app. The Hero Job Board application's logic requires that by the time we click the **Invite heroes** button, we need to know 
+  the list of heroes to invite, so that the corresponding `HeroPanel` components could be nested into `HeroJobBoard`.
+  
+  For the sake of simplicity, we declare the list of heroes to invite as the content of `HeroJobBoard`:
+  
++makeExample('component-communication/ts/app/take-job-contentchildren/job-board-with-content.ts', null, 'take-job-contentchildren/job-board-with-content.ts', {blk: /(HeroCrm)/g})
+
+:marked
+.alert.is-helpful
+  :marked
+    We need a component that can be used as the one to be the root component displayed by `router-outlet` we use in this example &mdash; that is why we
```

Comment:
The fact that we use a router shouldn't come up at this level of detail. It's part of the sample harness ... the thing that enables us to jump among the examples within this chapter.

It plays no role in the example you're talking about.